File: ProblemSolve/1022.cs
/**************************************
 * Sum of Root To Leaf Binary Numbers *
 **************************************/

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public int SumRootToLeaf(TreeNode root, int curSum = 0) {
        if(root == null){
            return 0;
        }
        
        if(root.left == null && root.right == null){
            return 2 * curSum + root.val;
        }
        
        return SumRootToLeaf(root.left, 2 * curSum + root.val) + SumRootToLeaf(root.right, 2 * curSum + root.val);
    }
}

File: ProblemSolve/1022.cpp
/**************************************
 * Sum of Root To Leaf Binary Numbers *
 **************************************/

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    int sumRootToLeaf(TreeNode* root, int curSum = 0)  {        
        if(root == NULL){
            return 0;
        }
        
        //leaf인 경우 합 계산
        if(root -> left == NULL && root -> right == NULL){
            return 2 * curSum + root -> val;
        }
        
        //재귀적으로 sum 계산하면서 진행
        return sumRootToLeaf(root->left, 2 * curSum + root -> val) + sumRootToLeaf(root->right, 2 * curSum + root -> val);
    }
};